addons/survey/tools/preview_image.py

```diff
@@ -0,0 +1,90 @@
+from PIL import Image, ImageDraw, ImageEnhance, ImageFont
+
+from odoo import _
+from odoo.tools import format_decimalized_number, file_path
+
+FONT = ImageFont.truetype(file_path("web/static/fonts/google/Montserrat/Montserrat-SemiBold.ttf"), 32)
+FONT_BOLD = ImageFont.truetype(file_path("web/static/fonts/google/Montserrat/Montserrat-Bold.ttf"), 64)
+FONT_BOLD_LARGE = FONT_BOLD.font_variant(size=80)
+OPENGRAPH_DIMENSIONS = (1200, 630)
+PADDING = 80  # Padding around the preview's content
+SECONDARY_IMAGE_DIMENSIONS = (40, 40)
+SPACER = 12  # Spacing between elements
+
+def generate_survey_preview_image(
+    background_image, nb_completed, nb_registered, secondary_image, secondary_text, success_rate,
+    tag_description, tag_path, title):
+    """Create the image used inside link previews.
+    The image is created on the fly so that we don't need to store it.
+    """
+
+    # 1200x630 are opengraph's advised dimensions
+    image = Image.new('RGBA', OPENGRAPH_DIMENSIONS, color="white")
+    d = ImageDraw.Draw(image)
+
+    # Use the background image as the preview's background if there is one
+    if background_image:
+        # Darken background image
+        enhancer = ImageEnhance.Brightness(background_image)
+        background_image = enhancer.enhance(0.4)
+        # Resize background image while keeping the ratio
+        background_image = background_image.resize((image.width, int(background_image.height * image.width / background_image.width)))
+        image.paste(background_image)
+        # Recolor texts to make them readable on darkened image
+        font_color = "#F8F7F7"
+        font_color_light = "#D4D1CD"
+        font_color_brand = font_color_light
+    else:
+        font_color = "#333333"
+        font_color_light = "#666666"
+        font_color_brand = "#714B67"
+        # Decorative Odoo logo's "O" (top right)
+        d.ellipse((image.width - 274, -274, image.width + 274, 274), fill=None, outline="#EFEBEE", width=112)
+
+    # Tag and title
+    icon = Image.open(file_path(tag_path)).convert("RGBA")
+    image.paste(icon, (PADDING, PADDING), icon)
+    d.text((PADDING + icon.width + SPACER, PADDING), tag_description, font=FONT, fill=font_color_brand)
+    d.text((PADDING, PADDING + icon.height + SPACER), wrap_title(title, image.width - 2 * PADDING), font=FONT_BOLD, fill=font_color)
+
+    # Secondary image and text
+    bottom_position_y = image.height - PADDING - FONT.size
```

Comment:
Shouldn't this be a `BOTTOM_TEXT_LINE_Y = OG_HEIGHT - PADDING - FONT.size` constant outside the function?

This allows to differentiate what is constant/static from what depends of the function inputs.